### PR TITLE
Fix a few issues regarding the language modeling script

### PR DESCRIPTION
**Pull request by LysandreJik**
The language modeling script currently has a few issues.

- in the line by line dataset, no special tokens are added (that's due to the fact the `batch_encode_plus` has the `add_special_token` flag `False` by default, which is misleading).
- the max length is ill computed in that same dataset, as it doesn't take into account the fact that `encode_plus` is aware of the special tokens and their impact on the sequence length.

### CODE REVIEW

File: examples/run_language_modeling.py [hunk 1]
```diff
@@ -130,9 +130,9 @@ def __init__(self, tokenizer: PreTrainedTokenizer, args, file_path: str, block_s
         logger.info("Creating features from dataset file at %s", file_path)
 
         with open(file_path, encoding="utf-8") as f:
-            lines = [line for line in f.read().splitlines() if len(line) > 0]
+            lines = [line for line in f.read().splitlines() if (len(line) > 0 and not line.isspace())]
```
Comment (julien-c) [inline examples/run_language_modeling.py]: First sub-clause of the if clause is probably covered by the second sub-clause, not?

Comment (LysandreJik) [reply examples/run_language_modeling.py]: Unfortunately not, as tokenizing the wikitext-2 yields 1 character strings containing a blank space.

Comment (LysandreJik) [reply examples/run_language_modeling.py]: Ah, I misunderstood your comment but `''.isspace()` returns `False` so both clauses are necessary

Comment (julien-c) [reply examples/run_language_modeling.py]: I see, I was mistaken, thank you

File: examples/run_language_modeling.py [hunk 2]
```diff
@@ -195,6 +198,12 @@ def _rotate_checkpoints(args, checkpoint_prefix="checkpoint", use_mtime=False) -
 
 def mask_tokens(inputs: torch.Tensor, tokenizer: PreTrainedTokenizer, args) -> Tuple[torch.Tensor, torch.Tensor]:
     """ Prepare masked tokens inputs/labels for masked language modeling: 80% MASK, 10% random, 10% original. """
+
+    if tokenizer.mask_token is None:
+        raise ValueError(
+            "This tokenizer does not have a mask token which is necessary for masked language modeling. Remove the --mlm flag if you want to use this tokenizer."
+        )
```
Comment (LysandreJik) [inline examples/run_language_modeling.py]: The script was throwing a dark error when using gpt2 with `--mlm`, this error should make things clearer.

**Pull request merged**